### PR TITLE
1.4/1656 prevent scan page from allowing keyboard input

### PR DESCRIPTION
**Pull request by AlinChindea**
Trello 🎫 🔗 [https://trello.com/c/wfWZBUCf/1656-prevent-scan-page-from-allowing-keyboard-input-vvv-ee](https://trello.com/c/wfWZBUCf/1656-prevent-scan-page-from-allowing-keyboard-input-vvv-ee)

### CODE REVIEW

File: src/pages/Scan.vue [hunk 1]
```diff
@@ -21,19 +21,21 @@
...
                             type="text"
                             v-model="sponsorCode"
                             ref="sponsorBox"
                             minlength="2"
                             maxlength="5"
                             autofocus="autofocus"
                             v-bind:class="{ 'input-text-hidden': queued }"
-                        />
-                        <input
+                        >
```
Comment (AlinChindea) [inline src/pages/Scan.vue]:
```suggestion
                        />
```
ahhhh

File: src/pages/Scan.vue [hunk 2]
```diff
@@ -258,11 +263,21 @@ export default {
...
             if (char.match(rxNumber)) {
+                if (!TIMER) {
+                    this.delay(() => {
+                        // When the box gets full, cancel any timers, else remove input
+                        this.voucherCode.length === voucherBoxMaxLength ? TIMER = null : this.voucherCode = '';
+                        TIMER = null;
+                    }, 150);
+
+                }
                 if (this.voucherCode.length < event.target.maxlength) {
```
Comment (charlesstrange2) [inline src/pages/Scan.vue]: remove these; redundant

**Comment by charlesstrange2** [comment]
Tests are failing because the browser test suite "types" like a human. I suggest
- a test for the typing version that fails, not succeeds
- a test that uses the typeteext with a speed options parameter 

https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/typetext.html

**Review by charlesstrange2 — APPROVED** [review]
okedoke!
⌨️

**Pull request merged**